pkg/s3wrapper/filesystem.go

```diff
@@ -309,49 +302,57 @@ func (f *FSClient) ListObjectsByPrefix(ctx context.Context, prefix string) ([]st
 
 func (f *FSClient) UploadBootFiles(ctx context.Context, openshiftVersion, serviceBaseURL string, haveLatestMinimalTemplate bool) error {
```

Comment:
I think it would be useful to document the logic/expectations of this function in a docstring. Would you mind doing that in this PR?

This function does several things, in a specific order, for a reason. Having the docs would make it easier for future editors of this function to make sure the logic is not broken.